namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -80,6 +82,19 @@ private[consul] object SvcAddr {
             )
             stopped = true
             Future.Unit
+            // this exception case checks if we queried for a service in a datacenter that
+            // doesn't exist. We capture this case so that we can return Addr.Neg to prevent
+            // service name resolution from timing out.
+          case Throw(e: UnexpectedResponse) if e.rsp.contentString == DatacenterErrorMessage =>
```

Comment:
maybe include a link to the consul issue you filed in a comment here, as justification for why we need to do this.